modules/solace/examples_test.go

```diff
@@ -0,0 +1,214 @@
+		}),
+		sc.WithShmSize(1<<30),
+	)
+	defer func() {
+		if err := testcontainers.TerminateContainer(ctr); err != nil {
+			log.Printf("failed to terminate container: %s", err)
+		}
+	}()
+	fmt.Println(err)
+	// }
+
+	// Output:
+	// <nil>
+}
+
+func ExampleRun_withTopicAndQueue() {
+	ctx := context.Background()
+
+	ctr, err := sc.Run(ctx, "solace/solace-pubsub-standard:latest",
+		sc.WithCredentials("admin", "admin"),
+		sc.WithVPN("test-vpn"),
+		sc.WithServices(sc.ServiceAMQP, sc.ServiceManagement, sc.ServiceSMF),
+		testcontainers.WithEnv(map[string]string{
+			"username_admin_globalaccesslevel": "admin",
+			"username_admin_password":          "admin",
+		}),
+		sc.WithShmSize(1<<30),
+		sc.WithQueue("TestQueue", "Topic/MyTopic"),
+	)
+	defer func() {
+		if err := testcontainers.TerminateContainer(ctr); err != nil {
+			log.Printf("failed to terminate container: %s", err)
+		}
+	}()
+	fmt.Println(err)
+	// the [testMessagePublishAndConsume] function is responsible for printing the output
+	// to the console, so be aware of that when adding it to other examples.
+	err = testMessagePublishAndConsume(ctr, "TestQueue", "Topic/MyTopic")
+	fmt.Println(err)
+
+	// Output:
+	// <nil>
+	// Published message to topic: Topic/MyTopic
+	// Received message: Hello from Solace testcontainers!
+	// Successfully received message from queue: TestQueue
+	// <nil>
+}
+
+func testMessagePublishAndConsume(ctr *sc.Container, queueName, topicName string) error {
+	// Get the SMF service URL from the container
+	smfURL, err := ctr.BrokerURLFor(context.Background(), sc.ServiceSMF)
+	if err != nil {
+		return fmt.Errorf("failed to get SMF URL: %w", err)
+	}
+
+	// Configure connection properties
+	brokerConfig := config.ServicePropertyMap{
+		config.TransportLayerPropertyHost:                 smfURL,
+		config.ServicePropertyVPNName:                     ctr.VPN(),
+		config.AuthenticationPropertyScheme:               config.AuthenticationSchemeBasic,
+		config.AuthenticationPropertySchemeBasicUserName:  ctr.Username(),
+		config.AuthenticationPropertySchemeBasicPassword:  ctr.Password(),
+		config.TransportLayerPropertyReconnectionAttempts: 0,
+	}
+
+	// Build messaging service
+	messagingService, err := messaging.NewMessagingServiceBuilder().
+		FromConfigurationProvider(brokerConfig).
+		Build()
+	if err != nil {
+		return fmt.Errorf("failed to build messaging service: %w", err)
+	}
+
+	// Connect to the messaging service
+	if err := messagingService.Connect(); err != nil {
+		return fmt.Errorf("failed to connect to messaging service: %w", err)
+	}
+	defer func() {
+		if err := messagingService.Disconnect(); err != nil {
+			log.Printf("Error disconnecting from messaging service: %v", err)
+		}
+	}()
+
+	// Test message publishing
+	err = publishTestMessage(messagingService, topicName)
+	if err != nil {
+		return fmt.Errorf("failed to publish message: %w", err)
+	}
+
+	// Wait a moment for message to be delivered to queue
+	time.Sleep(1 * time.Second)
```

Comment:
question: is there any way of removing this time.Sleep? This value could change depending on the host running the tests

Reply:
I think i can use a channel to wait, would me more consistent